File: server/models/book.js
const mongoose = require("mongoose");

const bookSchema = mongoose.Schema({
  img: {
    type: String,
  },
  title: {
    required: true,
    type: String,
    trim: true,
  },
  authName: {
    required: true,
    type: String,
    trim: true,
  },
  genre: { 
    type: [String], 
    required: true 
  },
  price: {
    required: true,
    type: Number,
  },
  pageNum: {
    required: true,
    type: Number,
  },
  description: {
    required: true,
    type: String,
    trim: true,
  },
});

// to convert a schema to a model
const Book = mongoose.model("Book", bookSchema);
module.exports = Book;

File: lib/models/book.dart
import 'dart:convert';

class Book {
  final String img;
  final String id;
  final String title;
  final String authName;
  final List<String> genre;
  final double price;
  final int pageNum;
  final String description;

  Book({
    required this.img,
    required this.id,
    required this.title,
    required this.authName,
    required this.genre,
    required this.price,
    required this.pageNum,
    required this.description,
  });
  
  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['_id'],
      img: json['img'],
      title: json['title'],
      authName: json['authName'],
      genre: List<String>.from(json['genre']),
      price: json['price'].toDouble(),
      pageNum: json['pageNum'],
      description: json['description'],
    );
  }
}
